src/azure_devtools/scenario_tests/base.py

```diff
@@ -122,8 +122,12 @@ def __init__(self,  # pylint: disable=too-many-arguments
             recording_dir,
             '{}.yaml'.format(recording_name or method_name)
         )
-        if self.is_live and os.path.exists(self.recording_file):
-            os.remove(self.recording_file)
+        self.temp_recording_file = os.path.join(
+            recording_dir,
+            '{}.temp.yaml'.format(recording_name or method_name)
+        )
+        if self.is_live and os.path.exists(self.temp_recording_file):
+            os.remove(self.temp_recording_file)
```

Comment:
Maybe we can move VCR setup code to `setUp`? Not verified. 🤔

Reply:
That will change a lot of code. But you can have a try.